### PR TITLE
[3/3] Refactors /core controllers using constructor property promotion.

### PR DESCRIPTION
**Pull request by fsamapoor**
This PR is the third part of the three-part refactoring of the `/core` controllers by using PHP8's constructor property promotion to remove redundant lines and make the code cleaner.

I figured I should split the changes into three parts to make reviewing the changes easier.

### CODE REVIEW

File: core/Controller/SearchController.php [hunk 1]
```diff
@@ -34,19 +34,13 @@
...
 class SearchController extends Controller {
-	private ISearch $searcher;
-	private LoggerInterface $logger;
-
 	public function __construct(
 		string $appName,
 		IRequest $request,
-		ISearch $search,
-		LoggerInterface $logger
+		private ISearch $searcher,
```
Comment (artonge) [inline core/Controller/SearchController.php:40]: The rename might break DI, can you check that this does not change anything?

File: core/Controller/WellKnownController.php [hunk 2]
```diff
@@ -33,13 +33,11 @@
...
 
 class WellKnownController extends Controller {
-	/** @var RequestManager */
-	private $requestManager;
-
-	public function __construct(IRequest $request,
-								RequestManager $wellKnownManager) {
+	public function __construct(
+		IRequest $request,
+		private RequestManager $requestManager,
```
Comment (artonge) [inline core/Controller/WellKnownController.php:38]: Same, the rename might break DI.

File: core/Controller/SearchController.php (hunk 1 above)
Comment (fsamapoor) [reply core/Controller/SearchController.php:40]: Sure. Let me investigate and I'll keep you updated.

Comment (fsamapoor) [reply core/Controller/SearchController.php:40]: I injected it as it is to a controller and it worked without any problems.

Furthermore, `ISearch` is [one of the documented type hints](https://docs.nextcloud.com/server/latest/developer_manual/basics/dependency_injection.html#predefined-core-services) that can be used to inject core services.

However, I'm totally fine with reverting the renamings. Honestly, this was not a goal of mine and I just renamed the parameters thinking the test suit will show if I have broken anything in the process.

**Review by artonge — APPROVED** [review]
LGTM

File: core/Controller/SetupController.php [hunk 3]
```diff
@@ -35,15 +35,10 @@
...
 
 class SetupController {
-	protected Setup $setupHelper;
 	private string $autoConfigFile;
 
-	/**
-	 * @param Setup $setupHelper
-	 */
-	public function __construct(Setup $setupHelper) {
+	public function __construct(protected Setup $setupHelper) {
```
Comment (come-nc) [inline core/Controller/SetupController.php:40]:
```suggestion
	public function __construct(
		protected Setup $setupHelper,
	) {
```

**Review by come-nc — APPROVED** [review]

**Pull request merged**